runtime/near-vm-runner/src/tests/cache.rs

```diff
@@ -71,21 +75,41 @@ fn make_cached_contract_call_vm(
     let promise_results = vec![];
     context.prepaid_gas = prepaid_gas;
     let code = ContractCode::new(code.to_vec(), None);
-    let runtime = vm_kind.runtime().expect("runtime has not been compiled");
-
+    let runtime = vm_kind.runtime(config).expect("runtime has not been compiled");
     runtime.run(
         &code,
         method_name,
         &mut fake_external,
         context.clone(),
-        &config,
         &fees,
         &promise_results,
         LATEST_PROTOCOL_VERSION,
         Some(cache),
     )
 }
 
+#[test]
+fn test_wasmer2_artifact_output_stability() {
+    // If this test has failed, you want to adjust the necessary constants so that `cache::vm_hash`
+    // changes (and only then the hashes here).
+    //
+    // Note that this test is a best-effort fish net. Some changes that should modify the hash will
+    // fall through the cracks here, but hopefully it should catch most of the fish just fine.
+    let mut hasher = StableHasher::new();
+    let code = near_test_contracts::trivial_contract();
+    let config = VMConfig::test();
+    let prepared_code = prepare::prepare_contract(code, &config).unwrap();
+    let mut features = CpuFeature::set();
+    features.insert(CpuFeature::AVX);
+    let triple = "x86_64-unknown-linux-gnu".parse().unwrap();
+    let target = Target::new(triple, features);
+    let vm = Wasmer2VM::new_for_target(config, target);
+    let artifact = vm.compile_uncached(&prepared_code).unwrap();
+    let serialized = artifact.artifact().serialize().unwrap();
+    serialized.hash(&mut hasher);
+    assert_eq!(hasher.finish(), 7017891230624373240, "WASMER2_CONFIG needs version change");
```

Comment:
:+1: that's a very good test, thanks for realizing we can do this. It did happen in the past that we've *almost* forgot to bust the caches.